packages/pymedphys_labs/src/pymedphys_labs/pedromartinez/qc-jaws.py

```diff
@@ -0,0 +1,1072 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+
+###########################################################################################
+#
+#   Script name: qc-jaws
+#
+#   Description: Tool for calculating jaws junction shifts for linear accelerators.
+#   The script opens every DICOM file in a given folder and creates a combined profile
+#   resulting from the superposition of the two or more fields. It then detects the
+#   peak/through formed by the gap/overlap of the fields. A window is then selected
+#   around this point and a Savitzky-Golay smoothing filter is then applied to the
+#   combined profile. This new curve is then used iteratively to minimize the
+#   profile created every time one of the profiles slide to close the gap or decrease
+#   the overlap. The profile will achieve it greatest level of homogeneity when the
+#   dosimetric penumbra of both fields are matched in space. The final result is the
+#   optimal calculation of the gap/overlap between the two profiles. The software
+#   generates a pdf file with a summary of all the results.
+#
+#   Example usage: python qc-jaws "/folder/"
+#
+#   Author: Pedro Martinez
+#   pedro.enrique.83@gmail.com
+#   5877000722
+#   Date:2019-04-09
+#
+###########################################################################################
+
+import os
+import sys
+
+# sys.path.append('C:\Program Files\GDCM 2.8\lib')
+import pydicom
+from PIL import *
+import subprocess
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+from tqdm import tqdm
+import numpy as np
+from scipy.signal import butter, filtfilt
+from scipy.signal import savgol_filter
+from scipy.stats import linregress
+from scipy import signal
+from scipy.signal import find_peaks, peak_prominences, peak_widths
+import fitz
+
+
+
+def running_mean(x, N):
+    out = np.zeros_like(x, dtype=np.float64)
+    dim_len = x.shape[0]
+    for i in range(dim_len):
+        if N % 2 == 0:
+            a, b = i - (N - 1) // 2, i + (N - 1) // 2 + 2
+        else:
+            a, b = i - (N - 1) // 2, i + (N - 1) // 2 + 1
+
+        # cap indices to min and max indices
+        a = max(0, a)
+        b = min(dim_len, b)
+        out[i] = np.mean(x[a:b])

```

Comment:
Let's add the following comment to this line to ignore that pylint error until the respective bug fix comes from numpy.

```suggestion
        out[i] = np.mean(x[a:b])  # pylint: disable=unsupported-assignment-operation
```

Reply:
If you're happy with that change just press the following button :)

![image](https://user-images.githubusercontent.com/6559099/59074842-bde69c80-8910-11e9-83a5-e29ba9699bc2.png)